### PR TITLE
Fix non-chr/numeric casting in `apply_labs_haven()`

### LINKED ISSUE #189: [BUG] Test error with dev version of labelled
# Expected Behavior

Labelled version 2.13.0 is about to be submitted to CRAN.

In this new version, a bug has been fixed in set_value_labels(). The function now calls systematically haven::labelled() and report an error when trying to assign not accepted values.

# Current Behavior

With the dev version of labelled, some tests of REDCapTidier are failing. cf. https://win-builder.r-project.org/incoming_pretest/labelled_2.13.0_20240405_134651/reverseDependencies/summary.txt

After exploring, it seems to occur here in `test-utils.R`:

```
out <- multi_choice_to_labels(
    db_data = db_data_classic,
    db_metadata = db_metadata_classic,
    raw_or_label = "haven"
  ) %>%
    suppressWarnings(classes = c(
      "empty_parse_warning",
      "field_missing_categories",
      "duplicate_labels"
    ))
```

Trying to understand when it was coming from, it seems related to the variable `"radio_dtxt_error"` in `db_metadata_classic`.

Variable `db_data$radio_dtxt_error` contains only NA and is of class logical. And it has no value labels in metadata. Therefore the identified labels is only NA and of class logical.

If you try to apply value labels to a logical vector, you will get an error from haven because value labels could be applied only to numeric or to character vectors.

In `apply_labs_haven()`, I see these lines of code:

```
  if (!is.null(attr(labels_cast, "problems"))) {
    # If there was parsing problem fall back to character
    x <- as.character(x)
    labels_cast <- force_cast(labels, character())
  }
```

In this specific case, this condition is not triggerred. In this specific case, it would be important to convert x to character or numeric before calling `set_value_labels()`. (in current CRAN version of labelled, it doesn't produce an error while it should have done).

It seems that changing the code to 

```
  if (!is.null(attr(labels_cast, "problems")) || is.logical(x)) {
    # If there was parsing problem fall back to character
    x <- as.character(x)
    labels_cast <- force_cast(labels, character())
  }
```
 
is enough to solve the bug.

Would you like to consider such change?
Could you check if all tests pass?

If yes, would you consider to submit a new version on CRAN before I resubmit the dev version of labelled on CRAN?

## How to Reproduce the Bug:

Install dev version of labelled and run tests of REDCapTidier.

## Failure Logs

[If applicable, submit any failure logs or error messages here.](https://win-builder.r-project.org/incoming_pretest/labelled_2.13.0_20240405_134651/reverseDependencies/summary.txt)

### Checklist
Before submitting this issue, please check and verify below that the submission meets the below criteria:

- [ ] The issue is atomic
- [ ] The issue description is documented
- [ ] The issue title describes the problem succinctly
- [ ] Developers are assigned to the issue
- [ ] Labels are assigned to the issue

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR fixes test failures occurring against the development version of `labelled()`.

My initial implementation of `apply_labs_haven()` was more general than it needed to be and introduced a test failure on a radio button variable where both the data values and labels are missing and therefore `NA_logical`. [`haven::labelled()`](https://haven.tidyverse.org/reference/labelled.html) explicitly accepts only `character` and `numeric` data types so we don't actually have to account for data values of other data types and can just cast to `character` when they occur.

I also removed a failing test for a specific error message when the API returns status code 405. We were hitting google to generate the status code and it now returns 404. Not sure it's worth testing these.

# Proposed Changes 
- Remove failing status 405 test case
- Update `force_cast()` and `apply_labs_haven()` to fall back to converting to `character` when non-numeric data types are encountered

### Issue Addressed
closes #189 

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR
- [x] Pre-release package version incremented using `usethis::use_version()`

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [x] I checked that new files follow naming conventions and are in the right place
- [x] I checked that documentation is complete, clear, and without typos
- [x] I added/edited comments to explain "why" not "how"
- [x] I checked that all new variable and function names follow naming conventions
- [x] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,8 +1,16 @@
-# REDCapTidieR 1.1.0
+# REDCapTidieR 1.1.1 (development version)
 
 Version 1.1.0
```
Comment (rsh52) [inline NEWS.md:3]:
```suggestion
Version 1.1.1
```

File: tests/testthat/test-read_redcap.R [hunk 2]
```diff
@@ -370,7 +370,6 @@ test_that("read_redcap errors with bad inputs", {
   # redcap uri
   expect_error(read_redcap(123, Sys.getenv("REDCAPTIDIER_CLASSIC_API")), class = "check_character")
   expect_error(read_redcap(letters[1:3], Sys.getenv("REDCAPTIDIER_CLASSIC_API")), class = "check_character")
-  expect_error(read_redcap("https://www.google.com", Sys.getenv("REDCAPTIDIER_CLASSIC_API")), class = "cannot_post")
```
Comment (rsh52) [inline tests/testthat/test-read_redcap.R:373]: I think this should be ok, the next test tests an unresolvable host and we test anything that _would_ give us a 405 is handled correctly here:

https://github.com/CHOP-CGTInformatics/REDCapTidieR/blob/139aee878016470cde51d47a90fa11eff6ff3922/tests/testthat/test-utils.R#L315-L316

**Review by rsh52 — APPROVED** [review]
Two small comments, otherwise looks good. 👍
